### PR TITLE
[DOC] Fix mathematical representation on interval (range)

### PR DESCRIPTION
**Pull request by aeroastro**
I have fixed mathematical representation on intervals of histogram.

* Fix representation on range.
* Add additional description on the range.
* Remove ambiguity on example description.

cf.
https://en.wikipedia.org/wiki/Interval_(mathematics)#Including_or_excluding_endpoints

### CODE REVIEW

**Comment by s1monw** [comment]
@elasticmachine ok to test

File: docs/reference/aggregations/bucket/histogram-aggregation.asciidoc [hunk 1]
```diff
@@ -14,7 +14,8 @@ To make this more formal, here is the rounding function that is used:
 bucket_key = Math.floor((value - offset) / interval) * interval + offset
 --------------------------------------------------
 
-The `interval` must be a positive decimal, while the `offset` must be a decimal in `[0, interval[`.
+The `interval` must be a positive decimal, while the `offset` must be a decimal in `[0, interval)`
+(a decimal greater than or euqal to `0` and less than `interval`)
```
Comment (DaveCTurner) [inline docs/reference/aggregations/bucket/histogram-aggregation.asciidoc]: Typo: "equal" not "euqal"

File: docs/reference/aggregations/bucket/histogram-aggregation.asciidoc [hunk 2]
```diff
@@ -266,10 +267,10 @@ The above will sort the buckets based on the avg rating among the promoted produ
 ==== Offset
 
 By default the bucket keys start with 0 and then continue in even spaced steps of `interval`, e.g. if the interval is 10 the first buckets
-(assuming there is data inside them) will be [0 - 9], [10-19], [20-29]. The bucket boundaries can be shifted by using the `offset` option.
+(assuming there is data inside them) will be [0, 10), [10, 20), [20, 30). The bucket boundaries can be shifted by using the `offset` option.
 
-This can be best illustrated with an example. If there are 10 documents with values ranging from 5 to 14, using interval `10` will result in
-two buckets with 5 documents each. If an additional offset `5` is used, there will be only one single bucket [5-14] containing all the 10
+This can be best illustrated with an example. If there are 10 documents with integer values ranging from 5 to 14, using interval `10` will result in
```
Comment (DaveCTurner) [inline docs/reference/aggregations/bucket/histogram-aggregation.asciidoc]: This sentence is true even if the values are not integers.

File: docs/reference/aggregations/bucket/histogram-aggregation.asciidoc [hunk 3]
```diff
@@ -266,10 +267,10 @@ The above will sort the buckets based on the avg rating among the promoted produ
 ==== Offset
 
 By default the bucket keys start with 0 and then continue in even spaced steps of `interval`, e.g. if the interval is 10 the first buckets
-(assuming there is data inside them) will be [0 - 9], [10-19], [20-29]. The bucket boundaries can be shifted by using the `offset` option.
+(assuming there is data inside them) will be [0, 10), [10, 20), [20, 30). The bucket boundaries can be shifted by using the `offset` option.
```
Comment (DaveCTurner) [inline docs/reference/aggregations/bucket/histogram-aggregation.asciidoc]: For consistency, can you format these ranges as code (i.e. within `backticks`) please?

File: docs/reference/aggregations/bucket/histogram-aggregation.asciidoc [hunk 4]
```diff
@@ -266,10 +267,10 @@ The above will sort the buckets based on the avg rating among the promoted produ
...
 
-This can be best illustrated with an example. If there are 10 documents with values ranging from 5 to 14, using interval `10` will result in
-two buckets with 5 documents each. If an additional offset `5` is used, there will be only one single bucket [5-14] containing all the 10
+This can be best illustrated with an example. If there are 10 documents with integer values ranging from 5 to 14, using interval `10` will result in
+two buckets with 5 documents each. If an additional offset `5` is used, there will be only one single bucket [5, 15) containing all the 10
```
Comment (DaveCTurner) [inline docs/reference/aggregations/bucket/histogram-aggregation.asciidoc]: For consistency, can you format this range as code (i.e. within `backticks`) please?

**Comment by DaveCTurner** [comment]
Thanks @aeroastro - as a mathematician I like this change. I left a handful of comments.

@elasticmachine please test again but with feeling this time.

**Comment by aeroastro** [comment]
@DaveCTurner 
Thank you for your quick and detailed review!
I have fixed the document.

**Comment by DaveCTurner** [comment]
Ok, I suspect the CI build is failing because your branch is based off 40fe7865277bc01a14343371a1cb4b34db5baa62 from back in December. Could you merge the current master into your branch and we'll try again?

**Comment by aeroastro** [comment]
@DaveCTurner 
Thank you for your advice. I've merged the current master, and now the CI build is successful.

**Comment by DaveCTurner** [comment]
LGTM, but it's not an area I'm 100% familiar with so asking @colings86 to glance over it too. Thanks again @aeroastro.

**Review by colings86 — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
Thanks for the contribution @aeroastro!

**Comment by aeroastro** [comment]
Thank you 🐱

**Pull request merged**